ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -102,21 +123,37 @@ def _peer_groups(self) -> list | None:
 
                 peer_groups.append(mpls_peer_group)
 
-            if self.shared_utils.overlay_evpn_vxlan is True:
+            # TODO - not great to have this logic with not WAN
+            if self.shared_utils.overlay_evpn_vxlan is True and not self.shared_utils.wan_role:
                 # EVPN OVERLAY peer group - also in EBGP..
-                ebgp_peer_group = {
+                ibgp_peer_group = {
                     **self._generate_base_peer_group("evpn", "evpn_overlay_peers"),
                     "remote_as": self.shared_utils.bgp_as,
                 }
 
                 if self.shared_utils.evpn_role == "server":
-                    ebgp_peer_group["route_reflector_client"] = True
+                    ibgp_peer_group["route_reflector_client"] = True
 
-                peer_groups.append(ebgp_peer_group)
+                peer_groups.append(ibgp_peer_group)
 
             if self._is_mpls_server is True:
                 peer_groups.append({**self._generate_base_peer_group("mpls", "rr_overlay_peers"), "remote_as": self.shared_utils.bgp_as})
 
+            if self.shared_utils.wan_role == "server" and len(self._wan_route_reflectors) > 0:
```

Comment:
point for reviewer - here is we are both MPLS and WAN, the peer group is generated twice.